src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/SettingsHelper.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Data;
+using Roslyn.Utilities;
+using System.Linq;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.EditorConfigSettings.Data;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.EditorConfig.Features
+{
+    internal class SettingsHelper
+    {
+        public static ImmutableArray<IEditorConfigSettingInfo> GetSettingsSnapshots(Workspace workspace, string filePath)
+        {
+            var settingsAggregator = workspace.Services.GetRequiredService<ISettingsAggregator>();
+
+            var codeStyleProvider = settingsAggregator.GetSettingsProvider<CodeStyleSetting>(filePath);
+            var whitespaceProvider = settingsAggregator.GetSettingsProvider<WhitespaceSetting>(filePath);
+            var analyzerProvider = settingsAggregator.GetSettingsProvider<AnalyzerSetting>(filePath);
+
+            var codeStyleSnapshot = (IEditorConfigSettingInfo[])codeStyleProvider?.GetCurrentDataSnapshot().ToArray()!;
```

Comment:
we should gracefully handle when the settings provider is null
e.g.
```
var codeStyleSnapshot = codeStyleProvider?.GetCurrentDataSnapshot()?.SelectAsArray(s => (IEditorConfigSettingInfo)s) ?? ImmutableArray<IEditorConfigSettingInfo>.Empty;
```

and same for the rest